app/src/main/java/org/simple/clinic/deeplink/DeepLinkUpdate.kt

```diff
@@ -9,19 +9,27 @@ class DeepLinkUpdate : Update<DeepLinkModel, DeepLinkEvent, DeepLinkEffect> {
 
   override fun update(model: DeepLinkModel, event: DeepLinkEvent): Next<DeepLinkModel, DeepLinkEffect> {
     return when (event) {
-      is UserFetched -> {
-        val effect = if (event.user == null) {
-          NavigateToSetupActivity
+      is UserFetched -> userFetched(model, event)
+      is PatientFetched -> patientFetched(event)
+    }
+  }
+
+  private fun userFetched(
+      model: DeepLinkModel,
+      event: UserFetched
+  ): Next<DeepLinkModel, DeepLinkEffect> {
+    return when {
+      event.user?.loggedInStatus == User.LoggedInStatus.LOGGED_IN -> {
+        val effect = if (model.patientUuid != null) {
+          FetchPatient(model.patientUuid)
         } else {
```

Comment:
We don't require this `else`.
This can be simplified to :
```
     private fun userFetched(
      model: DeepLinkModel,
      event: UserFetched
  ): Next<DeepLinkModel, DeepLinkEffect> {
    val isUserLoggedIn = event.user?.loggedInStatus == User.LoggedInStatus.LOGGED_IN
    val effect = when {
      isUserLoggedIn && model.patientUuid != null -> FetchPatient(model.patientUuid)
      else -> NavigateToSetupActivity
    }

    return dispatch(effect)
  }

```